plugins/gateway/apigee/telemetry-policy.js

```diff
@@ -0,0 +1,117 @@
+// REQUEST
+//------>Header<------//
+var requestHeaderList = [];
+
+var requestHeaders = context.getVariable("request.headers.names"),
+
+requestResults = {};
+    
+requestHeaders = requestHeaders + '';
+
+requestHeaders = requestHeaders.slice(1, -1).split(', ');
+
+requestHeaders.forEach(function(k){
+  var v = context.getVariable("request.header." + k );
+  requestResults[k] = v;
+  var header = { key : k, value: v};
+  requestHeaderList.push(header);
+  
+});
+//------>Body<------//
+var reqContent = context.getVariable('request.content')
+
+if(reqContent !== null){
+    reqContent = crypto.base64(crypto.asBytes(reqContent))
+}
+
+
+var requestCommon = { 
+                        truncatedBody : true, 
+                        body: reqContent, 
+                        headers: requestHeaderList, 
+                        version: "0.0.1"
+                    };
+                    
+var req = { 
+            common : requestCommon, 
+            host: context.getVariable("request.header.host"), 
+            method: context.getVariable('request.verb'), 
+            path: context.getVariable("request.uri")
```

Comment:
Does `request.uri returns path with query? (we need to get path with query)

Reply:
yes it does